### PR TITLE
Add `normalize` ops

### PR DESCRIPTION
**Pull request by awsaf49**
$L_p$ Normalization is a popular operation available in both PyTorch and TensorFlow. Having it in Keras 3.0 would be a great addition. This PR will add the `normalize()` ops,

$$
v' = \frac{v}{\max(\lVert v \rVert_p, \epsilon)}
$$

This operation is widely used in the community; for example, `OpenAI-CLIP` or `Google-SigLIP` uses `l2_normalization` in their encoder embeddings.

## Ref:
* PyTorch: https://pytorch.org/docs/stable/_modules/torch/nn/functional.html#normalize
* TensorFlow: https://www.tensorflow.org/api_docs/python/tf/linalg/normalize

### CODE REVIEW

**Comment by fchollet** [comment]
Thanks for the PR. This is basically equivalent to `keras.utils.normalize` I think? If it's hard to find, maybe we can make it an op.

**Comment by awsaf49** [comment]
> Thanks for the PR. This is basically equivalent to `keras.utils.normalize` I think? If it's hard to find, maybe we can make it an op.

Yes, it is equivalent to `keras.utils.normalize`, somehow I missed it. Should I close this PR?

This op can also be used in `CLIP / SigLIP` example in keras.io (Issue: https://github.com/keras-team/keras-io/issues/1752)

**Comment by awsaf49** [comment]
> > Yes, it is equivalent to keras.utils.normalize, somehow I missed it. Should I close this PR?
> 
> Can you move the code to `ops/nn.py` and make an `Operation` class for it? Then we can double export it to `ops` and `utils` (for backwards compat). It actually makes more sense for it to be an op.

Sure, will do that

**Comment by awsaf49** [comment]
@fchollet I've moved the code to `ops/nn.py`, should I just add `keras.utils.normalize` in the `keras_export` here?

Also noticed quite a few operations like `regularizers`, `losses`, `metrics` uses `keras.utils.normalize` should I keep them intact or `keras.ops.normalize` ?

File: keras/ops/nn.py [hunk 1]
```diff
@@ -1840,3 +1840,57 @@ def ctc_loss(target, output, target_length, output_length, mask_index=0):
...
+
+@keras_export(
+    [
+        "keras.ops.normalize",
+        "keras.ops.nn.normalize",
+    ]
+)
+def normalize(x, order=2.0, axis=1, eps=1e-12):
+    """Perform Lp normalization of a tensor over the specified axis.
+        `v' = v / max(||v||_p, epsilon)`
```
Comment (fchollet) [inline keras/ops/nn.py:1867]: Use a code block and use Python syntax

File: keras/ops/nn.py [hunk 2]
```diff
@@ -1840,3 +1840,57 @@ def ctc_loss(target, output, target_length, output_length, mask_index=0):
...
+        return _normalize(x, self.order, self.axis, self.eps)
+
+
+@keras_export(
+    [
+        "keras.ops.normalize",
+        "keras.ops.nn.normalize",
+    ]
+)
+def normalize(x, order=2.0, axis=1, eps=1e-12):
```
Comment (fchollet) [inline keras/ops/nn.py:1865]: Use `axis=-1, order=2` for backwards compat.

`eps=1e-12` is not going to work in float32 or float16. It seems to assume float64. Just omit it from the signature and use `config.epsilon()` in the code.

File: keras/ops/nn.py (hunk 1 above)
Comment (awsaf49) [reply keras/ops/nn.py:1866]: Do you mean like this?
```
"""Perform Lp normalization of a tensor over the specified axis.
```python
        v' = v / max(||v||_p, epsilon)
```
Noticed `sigmoid()` uses following formatting,
https://github.com/keras-team/keras/blob/5bc8488c0ea3f43c70c70ebca919093cd56066eb/keras/activations/activations.py#L319-L322

Comment (fchollet) [reply keras/ops/nn.py:1866]: You can do:

```
It is defined as: `normalize(x) = x / max(norm(x), epsilon)`.
```

Comment (awsaf49) [reply keras/ops/nn.py]: Thanks. I've updated the code.

**Review by fchollet — COMMENTED** [review]
Thanks for the updates! Please add unit tests for shape inference and correctness in `ops/nn_tests.py`.

**Comment by awsaf49** [comment]
@fchollet I have added the tests.

**Review by fchollet — APPROVED** [review]
LGTM, thank you!
